packages/datadog-instrumentations/src/http/server.js

```diff
@@ -23,10 +25,38 @@ addHook({ name: 'https' }, http => {
 
 addHook({ name: 'http' }, http => {
   shimmer.wrap(http.ServerResponse.prototype, 'emit', wrapResponseEmit)
+  shimmer.wrap(http.ServerResponse.prototype, 'end', wrapWriteHead)
+  shimmer.wrap(http.ServerResponse.prototype, 'writeHead', wrapWriteHead)
+  shimmer.wrap(http.ServerResponse.prototype, 'write', wrapWriteHead)
   shimmer.wrap(http.Server.prototype, 'emit', wrapEmit)
   return http
 })
 
+function wrapWriteHead (writeHead) {
+  return function (statusCode) {
+    if (this.finished) return
```

Comment:
if the response was already sent, we need to block the call to avoid a "write after end" error